lib/helpers/classic-invocation-component.js

```diff
@@ -0,0 +1,37 @@
+function transformTagName(tagName) {
+  if (isNestedComponentTagName(tagName)) {
+    return transformNestedTagName(tagName);
+  }
+
+  return capitalizedTagName(tagName);
+}
+
+function isComponentTagName(tagName) {
```

Comment:
Components do **not** need dashes in their names (in either curly or angle forms) as of something around Ember 3.8ish. I think this needs to be tweaked...

Reply:
So thinking about this I'm not sure how to make this work without having this rule throwing a warning for every property in a template, e.g. `{{myRandomProperty}}`. At least when we're just looking for `-` and `/` components we know they are at least components or helpers?

I was initially thinking of adding another config option similar to `allow`, called `disallow` which will allow users to list ambiguously named components that they'd like to throw lint errors on.

Reply:
Just to clarify I use this helper in the lint rule here https://github.com/ember-template-lint/ember-template-lint/pull/768/files#diff-9ccffddebfd018d7d7d2aab10ffb3428R22

Reply:
> I was initially thinking of adding another config option similar to allow, called disallow which will allow users to list ambiguously named components that they'd like to throw lint errors on.

Yep, thats the sticky wicket for sure. I'd like to see this go a few different directions, lets do this:

* Ignore `-` for all `BlockStatement` invocations (other than the built in's which we can enumerage (`if`,`each`,`with`,`let`, ???)), we **know for sure** those are components 
* Add the `allow` / `disallow` to the config like you mentioned above to allow us to check (or ignore) specific tags
* Add another boolean config flag that allows you to choose the default value (do we only restrict things with a `-` or all "possible components")

Reply:
FYI sorry for the silence but I've working away on this when I get time. I'm almost finished all your suggested changes.